### PR TITLE
FIX Use Member::currentUser if exists instead of double checking for ID

### LINKED ISSUE #83: Fatal error: Call to a member function getTitle() on null
I'm getting this error.
I'm using PHP 5.6
Mac OSX
Apache 2.4
Silverstripe 3.6.1

I ran multiple flushes, with no change.
Uninstall and flush again, and the site works again.

```
Fatal error: Call to a member function getTitle() on null in .../debugbar/code/DebugBarSilverStripeCollector.php on line 174
--

1 | 0.0006 | 240496 | {main}( ) | ../main.php:0
2 | 4.7042 | 7525160 | Director::direct( ) | ../main.php:191
3 | 131.4706 | 23252328 | RequestProcessor->postRequest( ) | ../Director.php:183
4 | 131.4707 | 23252552 | DebugBarRequestFilter->postRequest( ) | ../RequestProcessor.php:42
5 | 131.4707 | 23252672 | DebugBar::renderDebugBar( ) | ../DebugBarRequestFilter.php:57
6 | 131.4707 | 23254512 | DebugBar\JavascriptRenderer->render( ) | ../DebugBar.php:226
7 | 131.4707 | 23254808 | DebugBar\JavascriptRenderer->getJsInitializationCode( ) | ../JavascriptRenderer.php:981
8 | 131.4708 | 23255144 | DebugBar\JavascriptRenderer->getJsControlsDefinitionCode( ) | ../JavascriptRenderer.php:1015
9 | 131.4710 | 23268080 | DebugBarSilverStripeCollector->getWidgets( ) | ../JavascriptRenderer.php:1059
```

Any ideas on how to fix?

### PR DESCRIPTION
**Pull request by robbieaverill**
Fixes #83

### CODE REVIEW

**Review by fspringveldt — APPROVED** [review]

**Comment by robbieaverill** [comment]
Ok, there was a PHP 5.3 incompatible change released in a minor version (1.14.0) of php-debugbar which is why the 5.3 build is failing. PR at https://github.com/maximebf/php-debugbar/pull/360.

Otherwise we can lock to `~1.13.0` of php-debugbar instead.

File: code/DebugBarSilverStripeCollector.php [hunk 1]
```diff
@@ -169,8 +169,7 @@ public function getWidgets()
 
         $userIcon = 'user-times';
         $userText = 'Not logged in';
-        if (Member::currentUserID()) {
-            $member = Member::currentUser();
+        if ($member = Member::currentUser()) {
```
Comment (NightJar) [inline code/DebugBarSilverStripeCollector.php:172]: While I personally don't disagree, isn't PSR-2 to have 1 statement per line?
i.e.
```php
$member = Member::currentUser();
if ($member) {
```
?

**Review by NightJar — COMMENTED** [review]
General code style question, over all a fine change.

File: code/DebugBarSilverStripeCollector.php (hunk 1 above)
Comment (robbieaverill) [reply code/DebugBarSilverStripeCollector.php:172]: I don't think there's anything in PSR-2 about that, both examples are valid and I think the differences are up to taste, perhaps.

We can't merge this pull request until https://github.com/maximebf/php-debugbar/pull/360 gets released thought - if you could request changes saying that that'd be nice.

**Review by NightJar — CHANGES_REQUESTED** [review]
This can't be merged until PR at maximebf/php-debugbar#360 is merged & released. At that point CI builds will start passing and we'll all be happier.

**Pull request merged**